// rmw-shell/lib/containers/AppLayout/AppLayout.js
'use strict';

exports.__esModule = true;
exports.AppLayout = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _materialUiResponsiveDrawer = require('material-ui-responsive-drawer');

var _Drawer = require('../../containers/Drawer');

var _reactRouterDom = require('react-router-dom');

var _Scrollbar = require('../../components/Scrollbar');

var _Scrollbar2 = _interopRequireDefault(_Scrollbar);

var _AppRoutes = require('../../components/AppRoutes');

var _AppRoutes2 = _interopRequireDefault(_AppRoutes);

var _withAppConfigs = require('../../withAppConfigs');

var _withAppConfigs2 = _interopRequireDefault(_withAppConfigs);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var AppLayout = exports.AppLayout = function (_Component) {
  _inherits(AppLayout, _Component);

  function AppLayout() {
    _classCallCheck(this, AppLayout);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  AppLayout.prototype.render = function render() {
    var _props = this.props,
        muiTheme = _props.muiTheme,
        history = _props.history,
        appConfig = _props.appConfig;

    var drawerWidth = appConfig.drawer_width;
    var path = history.location.pathname;
    var customRoutes = appConfig.routes ? appConfig.routes : [];
    var appRoutes = (0, _AppRoutes2.default)(appConfig.firebaseLoad);

    return _react2.default.createElement(
      'div',
      { style: { backgroundColor: muiTheme.palette.canvasColor, height: '100%' } },
      _react2.default.createElement(
        _materialUiResponsiveDrawer.ResponsiveDrawer,
        { width: drawerWidth },
        _react2.default.createElement(
          _Scrollbar2.default,
          null,
          _react2.default.createElement(_Drawer.DrawerHeader, null),
          _react2.default.createElement(_Drawer.DrawerContent, { path: path, history: history })
        )
      ),
      _react2.default.createElement(
        _reactRouterDom.Switch,
        null,
        customRoutes.map(function (Route, i) {
          return _react2.default.cloneElement(Route, { key: '@customRoutes/' + i });
        }),
        appRoutes.map(function (Route, i) {
          return _react2.default.cloneElement(Route, { key: '@appRoutes/' + i });
        })
      )
    );
  };

  return AppLayout;
}(_react.Component);

var mapStateToProps = function mapStateToProps(state) {
  var theme = state.theme,
      locale = state.locale,
      messaging = state.messaging;


  return {
    theme: theme, // We need this so the theme change triggers rerendering
    locale: locale,
    messaging: messaging
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps)((0, _muiThemeable2.default)()((0, _reactRouterDom.withRouter)((0, _withAppConfigs2.default)(AppLayout))));
