// rmw-shell/lib/components/Forms/RoleForm.js
'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactIntl = require('react-intl');

var _reduxForm = require('redux-form');

var _reduxFormMaterialUi = require('redux-form-material-ui');

var _reactList = require('react-list');

var _reactList2 = _interopRequireDefault(_reactList);

var _List = require('material-ui/List');

var _Subheader = require('material-ui/Subheader');

var _Subheader2 = _interopRequireDefault(_Subheader);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var RoleForm = function (_Component) {
  _inherits(RoleForm, _Component);

  function RoleForm() {
    _classCallCheck(this, RoleForm);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  RoleForm.prototype.render = function render() {
    var _this2 = this;

    var _props = this.props,
        handleSubmit = _props.handleSubmit,
        intl = _props.intl,
        initialized = _props.initialized,
        renderGrantItem = _props.renderGrantItem,
        grants = _props.grants;


    return _react2.default.createElement(
      'form',
      { onSubmit: handleSubmit, style: {
          height: '100%',
          alignItems: 'stretch',
          display: 'flex',
          flexWrap: 'wrap',
          justifyContent: 'center'
        } },
      _react2.default.createElement('button', { type: 'submit', style: { display: 'none' } }),
      _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          'div',
          null,
          _react2.default.createElement(_reduxForm.Field, {
            name: 'name',
            disabled: !initialized,
            component: _reduxFormMaterialUi.TextField,
            hintText: intl.formatMessage({ id: 'name_hint' }),
            floatingLabelText: intl.formatMessage({ id: 'name_label' }),
            ref: 'name',
            withRef: true
          })
        ),
        _react2.default.createElement(
          'div',
          null,
          _react2.default.createElement(_reduxForm.Field, {
            name: 'description',
            component: _reduxFormMaterialUi.TextField,
            disabled: !initialized,
            hintText: intl.formatMessage({ id: 'description_hint' }),
            floatingLabelText: intl.formatMessage({ id: 'description_label' }),
            multiLine: true,
            rows: 2,
            ref: 'description',
            withRef: true
          })
        )
      ),
      _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _Subheader2.default,
          null,
          intl.formatMessage({ id: 'grants' })
        ),
        _react2.default.createElement(
          _List.List,
          { style: { height: '100%' }, ref: function ref(field) {
              _this2.grants = field;
            } },
          _react2.default.createElement(_reactList2.default, {
            itemRenderer: renderGrantItem,
            length: grants ? grants.length : 0,
            type: 'simple'
          })
        )
      )
    );
  };

  return RoleForm;
}(_react.Component);

RoleForm.propTypes = process.env.NODE_ENV !== "production" ? {
  handleSubmit: _propTypes2.default.func.isRequired,
  renderGrantItem: _propTypes2.default.func.isRequired,
  intl: _reactIntl.intlShape.isRequired,
  initialized: _propTypes2.default.bool.isRequired,
  uid: _propTypes2.default.string
} : {};

exports.default = (0, _reduxForm.reduxForm)({ form: 'role' })(RoleForm);
module.exports = exports['default'];
