// rmw-shell/lib/containers/PublicChats/PublicChats.js
'use strict';

exports.__esModule = true;
exports.Chat = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _reactIntl = require('react-intl');

var _actions = require('../../store/simpleValues/actions');

var _Activity = require('../../containers/Activity');

var _Activity2 = _interopRequireDefault(_Activity);

var _reactRouterDom = require('react-router-dom');

var _firekitProvider = require('firekit-provider');

var _ChatMessages = require('../../containers/ChatMessages/ChatMessages');

var _ChatMessages2 = _interopRequireDefault(_ChatMessages);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Chat = exports.Chat = function (_Component) {
  _inherits(Chat, _Component);

  function Chat() {
    _classCallCheck(this, Chat);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  Chat.prototype.render = function render() {
    var _props = this.props,
        muiTheme = _props.muiTheme,
        intl = _props.intl,
        firebaseApp = _props.firebaseApp;


    return _react2.default.createElement(
      _Activity2.default,
      {
        containerStyle: { overflow: 'hidden', backgroundColor: muiTheme.chip.backgroundColor },
        title: intl.formatMessage({ id: 'public_chats' }) },
      _react2.default.createElement(_ChatMessages2.default, { path: 'public_chats', firebaseApp: firebaseApp })
    );
  };

  return Chat;
}(_react.Component);

Chat.propTypes = process.env.NODE_ENV !== "production" ? {
  intl: _reactIntl.intlShape.isRequired,
  muiTheme: _propTypes2.default.object.isRequired,
  auth: _propTypes2.default.object.isRequired
} : {};

var mapStateToProps = function mapStateToProps(state, ownPops) {
  var auth = state.auth;
  var match = ownPops.match;

  var uid = match.params.uid;

  return {
    uid: uid,
    auth: auth
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, { setSimpleValue: _actions.setSimpleValue })((0, _reactIntl.injectIntl)((0, _muiThemeable2.default)()((0, _reactRouterDom.withRouter)((0, _firekitProvider.withFirebase)(Chat)))));
