// rmw-shell/lib/containers/MyAccount/MyAccount.js
'use strict';

exports.__esModule = true;
exports.MyAccount = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactIntl = require('react-intl');

var _Activity = require('../../containers/Activity');

var _Activity2 = _interopRequireDefault(_Activity);

var _actions = require('../../store/simpleValues/actions');

var _MyAccountForm = require('../../components/Forms/MyAccountForm');

var _MyAccountForm2 = _interopRequireDefault(_MyAccountForm);

var _reactRouterDom = require('react-router-dom');

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _FlatButton = require('material-ui/FlatButton');

var _FlatButton2 = _interopRequireDefault(_FlatButton);

var _Dialog = require('material-ui/Dialog');

var firebase = require('firebase');

var _Dialog2 = _interopRequireDefault(_Dialog);

var _firekitProvider = require('firekit-provider');

var _fireform = require('fireform');

var _fireform2 = _interopRequireDefault(_fireform);

var _Icons = require('../../components/Icons');

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _reduxForm = require('redux-form');

var _materialUiResponsiveMenu = require('material-ui-responsive-menu');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var path = '/users/';
var form_name = 'my_account';

var MyAccount = exports.MyAccount = function (_Component) {
  _inherits(MyAccount, _Component);

  function MyAccount() {
    var _temp, _this, _ret;

    _classCallCheck(this, MyAccount);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.getProviderIcon = function (p) {
      var muiTheme = _this.props.muiTheme;

      var color = muiTheme.palette.primary2Color;

      switch (p) {
        case 'google.com':
          return _react2.default.createElement(_Icons.GoogleIcon, { color: color });

        case 'facebook.com':
          return _react2.default.createElement(_Icons.FacebookIcon, { color: color });

        case 'twitter.com':
          return _react2.default.createElement(_Icons.TwitterIcon, { color: color });

        case 'github.com':
          return _react2.default.createElement(_Icons.GitHubIcon, { color: color });

        default:
          return undefined;
      }
    }, _this.handleEmailVerificationsSend = function () {
      var firebaseApp = _this.props.firebaseApp;

      firebaseApp.auth().currentUser.sendEmailVerification().then(function () {
        alert('Verification E-Mail send');
      });
    }, _this.handlePhotoUploadSuccess = function (snapshot) {
      var _this$props = _this.props,
          setSimpleValue = _this$props.setSimpleValue,
          change = _this$props.change;

      change(form_name, 'photoURL', snapshot.downloadURL);
      setSimpleValue('new_company_photo', undefined);
    }, _this.handleUserDeletion = function () {
      var _this$props2 = _this.props,
          change = _this$props2.change,
          submit = _this$props2.submit;

      change(form_name, 'delete_user', true);
      submit(form_name);
    }, _this.getProvider = function (provider) {
      if (provider.indexOf('facebook') > -1) {
        return new firebase.auth.FacebookAuthProvider();
      }
      if (provider.indexOf('github') > -1) {
        return new firebase.auth.GithubAuthProvider();
      }
      if (provider.indexOf('google') > -1) {
        return new firebase.auth.GoogleAuthProvider();
      }
      if (provider.indexOf('twitter') > -1) {
        return new firebase.auth.TwitterAuthProvider();
      }
      if (provider.indexOf('phone') > -1) {
        return new firebase.auth.PhoneAuthProvider();
      }

      throw new Error('Provider is not supported!');
    }, _this.reauthenticateUser = function (values, onSuccess) {
      var _this$props3 = _this.props,
          auth = _this$props3.auth,
          firebaseApp = _this$props3.firebaseApp,
          authError = _this$props3.authError;


      if (_this.isLinkedWithProvider('password') && !values) {
        if (onSuccess && onSuccess instanceof Function) {
          onSuccess();
        }
      } else if (_this.isLinkedWithProvider('password') && values) {
        var credential = firebase.auth.EmailAuthProvider.credential(auth.email, values.old_password);
        firebaseApp.auth().currentUser.reauthenticateWithCredential(credential).then(function () {
          if (onSuccess && onSuccess instanceof Function) {
            onSuccess();
          }
        }, function (e) {
          authError(e);
        });
      } else {
        firebaseApp.auth().currentUser.reauthenticateWithPopup(_this.getProvider(auth.providerData[0].providerId)).then(function () {
          if (onSuccess && onSuccess instanceof Function) {
            onSuccess();
          }
        }, function (e) {
          authError(e);
        });
      }
    }, _this.isLinkedWithProvider = function (provider) {
      var auth = _this.props.auth;


      try {
        return auth && auth.providerData && auth.providerData.find(function (p) {
          return p.providerId === provider;
        }) !== undefined;
      } catch (e) {
        return false;
      }
    }, _this.linkUserWithPopup = function (provider) {
      var _this$props4 = _this.props,
          firebaseApp = _this$props4.firebaseApp,
          authError = _this$props4.authError,
          authStateChanged = _this$props4.authStateChanged;


      firebaseApp.auth().currentUser.linkWithPopup(_this.getProvider(provider)).then(function (payload) {
        authStateChanged(firebaseApp.auth().currentUser);
      }, function (e) {
        authError(e);
      });
    }, _this.handleCreateValues = function (values) {
      return false;
    }, _this.clean = function (obj) {
      Object.keys(obj).forEach(function (key) {
        return obj[key] === undefined && delete obj[key];
      });
      return obj;
    }, _this.handleUpdateValues = function (values, dispatch, props) {
      var _this$props5 = _this.props,
          auth = _this$props5.auth,
          firebaseApp = _this$props5.firebaseApp,
          authStateChanged = _this$props5.authStateChanged,
          authError = _this$props5.authError;


      var simpleChange = values.displayName && values.displayName.localeCompare(auth.displayName) || values.photoURL && values.photoURL.localeCompare(auth.photoURL);

      var simpleValues = {
        displayName: values.displayName,
        photoURL: values.photoURL

        //Change simple data
      };if (simpleChange) {
        firebaseApp.auth().currentUser.updateProfile(simpleValues).then(function () {

          firebaseApp.database().ref('users/' + auth.uid).update(_this.clean(simpleValues)).then(function () {
            authStateChanged(values);
          }, function (e) {
            authError(e);
          });
        }, function (e) {
          authError(e);
        });
      }

      //Change email
      if (values.email && values.email.localeCompare(auth.email)) {

        _this.reauthenticateUser(values, function () {
          firebaseApp.auth().currentUser.updateEmail(values.email).then(function () {
            firebaseApp.database().ref('users/' + auth.uid).update({ email: values.email }).then(function () {
              authStateChanged({ email: values.email });
            }, function (e) {
              authError(e);
            });
          }, function (e) {
            authError(e);

            if (e.code === 'auth/requires-recent-login') {
              firebaseApp.auth().signOut().then(function () {
                setTimeout(function () {
                  alert('Please sign in again to change your email.');
                }, 1);
              });
            }
          });
        });
      }

      //Change password
      if (values.new_password) {

        _this.reauthenticateUser(values, function () {
          firebaseApp.auth().currentUser.updatePassword(values.new_password).then(function () {
            firebaseApp.auth().signOut();
          }, function (e) {
            authError(e);

            if (e.code === 'auth/requires-recent-login') {
              firebaseApp.auth().signOut().then(function () {
                setTimeout(function () {
                  alert('Please sign in again to change your password.');
                }, 1);
              });
            }
          });
        });
      }

      // We manage the data saving above
      return false;
    }, _this.handleClose = function () {
      var setSimpleValue = _this.props.setSimpleValue;

      setSimpleValue('delete_user', false);
      setSimpleValue('auth_menu', false);
    }, _this.handleDelete = function () {
      var _this$props6 = _this.props,
          firebaseApp = _this$props6.firebaseApp,
          authError = _this$props6.authError;


      _this.reauthenticateUser(false, function () {
        firebaseApp.auth().currentUser.delete().then(function () {
          _this.handleClose();
        }, function (e) {
          authError(e);

          if (e.code === 'auth/requires-recent-login') {
            firebaseApp.auth().signOut().then(function () {
              setTimeout(function () {
                alert('Please sign in again to delete your account.');
              }, 1);
            });
          }
        });
      });
    }, _this.validate = function (values) {
      var auth = _this.props.auth;

      var providerId = auth.providerData[0].providerId;
      var errors = {};

      if (!values.displayName) {
        errors.displayName = 'Required';
      }

      if (!values.email) {
        errors.email = 'Required';
      } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
        errors.email = 'Invalid email address';
      } else if (!values.old_password && providerId === 'password' && auth.email.localeCompare(values.email)) {
        errors.old_password = 'For email change enter your password';
      }

      if (values.new_password) {
        if (values.new_password.length < 6) {
          errors.new_password = 'Password should be at least 6 characters';
        } else if (values.new_password.localeCompare(values.new_password_confirmation)) {
          errors.new_password = 'Must be equal';
          errors.new_password_confirmation = 'Must be equal';
        }
      }

      return errors;
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  MyAccount.prototype.render = function render() {
    var _props = this.props,
        history = _props.history,
        intl = _props.intl,
        setSimpleValue = _props.setSimpleValue,
        delete_user = _props.delete_user,
        auth = _props.auth,
        muiTheme = _props.muiTheme,
        submit = _props.submit,
        firebaseApp = _props.firebaseApp;


    var actions = [_react2.default.createElement(_FlatButton2.default, {
      label: intl.formatMessage({ id: 'cancel' }),
      primary: true,
      onClick: this.handleClose
    }), _react2.default.createElement(_FlatButton2.default, {
      label: intl.formatMessage({ id: 'delete' }),
      secondary: true,
      onClick: this.handleDelete
    })];

    var menuList = [{
      hidden: auth.uid === undefined,
      text: intl.formatMessage({ id: 'save' }),
      icon: _react2.default.createElement(
        _FontIcon2.default,
        { className: 'material-icons', color: muiTheme.palette.canvasColor },
        'save'
      ),
      tooltip: intl.formatMessage({ id: 'save' }),
      onClick: function onClick() {
        return submit('my_account');
      }
    }, {
      hidden: auth.uid === undefined,
      text: intl.formatMessage({ id: 'delete' }),
      icon: _react2.default.createElement(
        _FontIcon2.default,
        { className: 'material-icons', color: muiTheme.palette.canvasColor },
        'delete'
      ),
      tooltip: intl.formatMessage({ id: 'delete' }),
      onClick: function onClick() {
        return setSimpleValue('delete_user', true);
      }
    }];

    return _react2.default.createElement(
      _Activity2.default,
      {
        iconStyleRight: { width: '50%' },
        iconElementRight: _react2.default.createElement(_materialUiResponsiveMenu.ResponsiveMenu, {
          iconMenuColor: muiTheme.palette.canvasColor,
          menuList: menuList
        }),
        title: intl.formatMessage({ id: 'my_account' }) },
      auth.uid && _react2.default.createElement(
        'div',
        { style: { margin: 15, display: 'flex' } },
        _react2.default.createElement(
          _fireform2.default,
          {
            firebaseApp: firebaseApp,
            validate: this.validate,
            name: form_name,
            path: path,
            handleUpdateValues: this.handleUpdateValues,
            onSubmitSuccess: function onSubmitSuccess(values) {
              history.push('/dashboard');setSimpleValue('auth_menu', false);
            },
            onDelete: function onDelete(values) {
              history.push('/signin');
            },
            handleCreateValues: this.handleCreateValues,
            uid: auth.uid },
          _react2.default.createElement(_MyAccountForm2.default, _extends({
            linkUserWithPopup: this.linkUserWithPopup,
            isLinkedWithProvider: this.isLinkedWithProvider,
            getProviderIcon: this.getProviderIcon,
            handleEmailVerificationsSend: this.handleEmailVerificationsSend,
            handlePhotoUploadSuccess: this.handlePhotoUploadSuccess,
            handleUserDeletion: this.handleUserDeletion
          }, this.props))
        )
      ),
      _react2.default.createElement(
        _Dialog2.default,
        {
          title: intl.formatMessage({ id: 'delete_account_dialog_title' }),
          actions: actions,
          modal: false,
          open: delete_user === true,
          onRequestClose: this.handleClose },
        intl.formatMessage({ id: 'delete_account_dialog_message' })
      )
    );
  };

  return MyAccount;
}(_react.Component);

MyAccount.propTypes = process.env.NODE_ENV !== "production" ? {
  history: _propTypes2.default.object,
  setSimpleValue: _propTypes2.default.func.isRequired,
  intl: _reactIntl.intlShape.isRequired,
  isGranted: _propTypes2.default.func,
  auth: _propTypes2.default.object.isRequired,
  vehicle_types: _propTypes2.default.array
} : {};

var selector = (0, _reduxForm.formValueSelector)(form_name);

var mapStateToProps = function mapStateToProps(state) {
  var intl = state.intl,
      simpleValues = state.simpleValues,
      auth = state.auth;


  var delete_user = simpleValues.delete_user;
  var new_user_photo = simpleValues.new_user_photo;

  return {
    new_user_photo: new_user_photo,
    intl: intl,
    delete_user: delete_user,
    auth: auth,
    photoURL: selector(state, 'photoURL'),
    old_password: selector(state, 'old_password')
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, { setSimpleValue: _actions.setSimpleValue, change: _reduxForm.change, submit: _reduxForm.submit })((0, _reactIntl.injectIntl)((0, _reactRouterDom.withRouter)((0, _muiThemeable2.default)()((0, _firekitProvider.withFirebase)(MyAccount)))));
