// rmw-shell/lib/containers/Activity/Activity.js
'use strict';

exports.__esModule = true;
exports.Activity = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

var _materialUiResponsiveDrawer = require('material-ui-responsive-drawer');

var _reactHelmet = require('react-helmet');

var _IconButton = require('material-ui/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _LinearProgress = require('material-ui/LinearProgress');

var _LinearProgress2 = _interopRequireDefault(_LinearProgress);

var _reactIntl = require('react-intl');

var _colors = require('material-ui/styles/colors');

var _reactRouterDom = require('react-router-dom');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Activity = exports.Activity = function (_Component) {
  _inherits(Activity, _Component);

  function Activity() {
    var _temp, _this, _ret;

    _classCallCheck(this, Activity);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.getIconElementLeft = function () {
      var onBackClick = _this.props.onBackClick;


      if (onBackClick) {
        return _react2.default.createElement(
          _IconButton2.default,
          { onClick: onBackClick },
          _react2.default.createElement(
            _FontIcon2.default,
            { className: 'material-icons' },
            'chevron_left'
          )
        );
      } else {
        return undefined;
      }
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  Activity.prototype.render = function render() {
    var _props = this.props,
        muiTheme = _props.muiTheme,
        title = _props.title,
        children = _props.children,
        onBackClick = _props.onBackClick,
        history = _props.history,
        intl = _props.intl,
        isConnected = _props.isConnected,
        isLoading = _props.isLoading,
        dispatch = _props.dispatch,
        containerStyle = _props.containerStyle,
        pageTitle = _props.pageTitle,
        height = _props.height,
        staticContext = _props.staticContext,
        valueLink = _props.valueLink,
        customDrawerWidth = _props.customDrawerWidth,
        rest = _objectWithoutProperties(_props, ['muiTheme', 'title', 'children', 'onBackClick', 'history', 'intl', 'isConnected', 'isLoading', 'dispatch', 'containerStyle', 'pageTitle', 'height', 'staticContext', 'valueLink', 'customDrawerWidth']);

    var drawerWidth = customDrawerWidth ? customDrawerWidth : 256;

    var bodyContainerStyle = _extends({
      backgroundColor: muiTheme.palette.canvasColor,
      top: 64,
      bottom: 0,
      overflow: 'auto'
    }, containerStyle);

    var headerTitle = '';

    if (typeof title === 'string' || title instanceof String) {
      headerTitle = title;
    }

    if (pageTitle) {
      headerTitle = pageTitle;
    }

    return _react2.default.createElement(
      'div',
      { style: { backgroundColor: muiTheme.palette.canvasColor, height: '100%' } },
      _react2.default.createElement(
        _reactHelmet.Helmet,
        null,
        _react2.default.createElement('meta', { name: 'theme-color', content: muiTheme.palette.primary1Color }),
        _react2.default.createElement('meta', { name: 'apple-mobile-web-app-status-bar-style', content: muiTheme.palette.primary1Color }),
        _react2.default.createElement('meta', { name: 'msapplication-navbutton-color', content: muiTheme.palette.primary1Color }),
        _react2.default.createElement(
          'title',
          null,
          headerTitle
        )
      ),
      _react2.default.createElement(_materialUiResponsiveDrawer.ResponsiveAppBar, _extends({
        width: drawerWidth,
        title: title,
        showMenuIconButton: onBackClick !== undefined ? true : undefined,
        onLeftIconButtonClick: onBackClick,
        iconElementLeft: this.getIconElementLeft()
      }, rest)),
      !isConnected && _react2.default.createElement(
        'div',
        {
          id: 'offline-indicator',
          style: {
            zIndex: 9999,
            position: 'fixed',
            top: 0,
            height: 12,
            backgroundColor: _colors.deepOrange500,
            color: _colors.darkWhite,
            width: '100%',
            fontSize: 12,
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center'

          } },
        _react2.default.createElement(
          'span',
          null,
          intl.formatMessage({ id: 'no_connection' })
        )
      ),
      isLoading && _react2.default.createElement(_LinearProgress2.default, { mode: 'indeterminate', color: muiTheme.palette.accent1Color, style: { zIndex: 9998, position: 'fixed', top: 0, height: height ? height : 5 } }),
      _react2.default.createElement(
        _materialUiResponsiveDrawer.BodyContainer,
        { width: drawerWidth, id: 'bodyContainer', ref: 'bodyContainer', withRef: true, style: bodyContainerStyle },
        children
      )
    );
  };

  return Activity;
}(_react.Component);

var mapStateToProps = function mapStateToProps(state) {
  var connection = state.connection,
      intl = state.intl;


  return {
    isConnected: connection ? connection.isConnected : false,
    intl: intl
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps)((0, _reactIntl.injectIntl)((0, _muiThemeable2.default)()((0, _reactRouterDom.withRouter)(Activity))));
