// rmw-shell/lib/components/PageNotFound/PageNotFound.js
'use strict';

exports.__esModule = true;

var _home = require('material-ui/svg-icons/action/home');

var _home2 = _interopRequireDefault(_home);

var _FloatingActionButton = require('material-ui/FloatingActionButton');

var _FloatingActionButton2 = _interopRequireDefault(_FloatingActionButton);

var _Paper = require('material-ui/Paper');

var _Paper2 = _interopRequireDefault(_Paper);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _withAppConfigs = require('../../withAppConfigs');

var _withAppConfigs2 = _interopRequireDefault(_withAppConfigs);

var _reactIntl = require('react-intl');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  paper: {
    height: '100%',
    margin: 0,
    padding: 1
  },
  container: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 5
  },
  icon: {
    width: 192,
    height: 192
  }
};

var PageNotFound = function PageNotFound(_ref) {
  var muiTheme = _ref.muiTheme,
      intl = _ref.intl,
      appConfig = _ref.appConfig;

  var AppIcon = appConfig.appIcon;

  return _react2.default.createElement(
    _Paper2.default,
    { zDepth: 1, style: styles.paper },
    _react2.default.createElement(
      'div',
      { style: styles.container },
      _react2.default.createElement(AppIcon, { color: muiTheme.palette.primary2Color, style: styles.icon }),
      _react2.default.createElement(
        'h3',
        null,
        intl.formatMessage({ id: 'warning_404_message' })
      ),
      _react2.default.createElement(
        'p',
        null,
        intl.formatMessage({ id: 'warning_404_description' })
      ),
      _react2.default.createElement(
        _FloatingActionButton2.default,
        { secondary: true, href: '/' },
        _react2.default.createElement(_home2.default, null)
      )
    )
  );
};

exports.default = (0, _reactIntl.injectIntl)((0, _muiThemeable2.default)()((0, _withAppConfigs2.default)(PageNotFound)));
module.exports = exports['default'];
