// rmw-shell/lib/config.js
'use strict';

exports.__esModule = true;

var _Icons = require('./components/Icons');

var _store = require('./store');

var _store2 = _interopRequireDefault(_store);

var _menuItems = require('./menuItems');

var _menuItems2 = _interopRequireDefault(_menuItems);

var _locales = require('./locales');

var _locales2 = _interopRequireDefault(_locales);

var _grants = require('./grants');

var _grants2 = _interopRequireDefault(_grants);

var _themes = require('./themes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var config = {
  firebase_config: {
    apiKey: "AIzaSyDR6-mGXKNnkJ0Sv7ufX8oNCGYk5UxVVr0",
    authDomain: "spotaproject.firebaseapp.com",
    databaseURL: "https://spotaproject.firebaseio.com",
    projectId: "spotaproject",
    storageBucket: "spotaproject.appspot.com",
    messagingSenderId: "746405070068"
  },
  firebase_config_dev: {
    apiKey: "AIzaSyDR6-mGXKNnkJ0Sv7ufX8oNCGYk5UxVVr0",
    authDomain: "spotaproject.firebaseapp.com",
    databaseURL: "https://spotaproject.firebaseio.com",
    projectId: "spotaproject",
    storageBucket: "spotaproject.appspot.com",
    messagingSenderId: "746405070068"
  },
  firebase_providers: ['google.com', 'facebook.com', 'twitter.com', 'github.com', 'password', 'phone'],
  initial_state: {
    theme: 'dark',
    locale: 'en'
  },
  drawer_width: 256,
  appIcon: _Icons.RMWIcon,
  configureStore: _store2.default,
  getMenuItems: _menuItems2.default,
  locales: _locales2.default,
  themes: _themes.themes,
  grants: _grants2.default,
  routes: [],
  onAuthStateChanged: undefined,
  firebaseLoad: function firebaseLoad() {
    return import('./firebase');
  }
};

exports.default = config;
module.exports = exports['default'];
