// rmw-shell/lib/components/SearchField/SearchField.js
'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _colorManipulator = require('material-ui/utils/colorManipulator');

var _reactIntl = require('react-intl');

var _TextField = require('material-ui/TextField');

var _TextField2 = _interopRequireDefault(_TextField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SearchField = function SearchField(_ref) {
  var onChange = _ref.onChange,
      hintText = _ref.hintText,
      muiTheme = _ref.muiTheme,
      intl = _ref.intl;


  return _react2.default.createElement(
    'div',
    { style: {
        display: 'inline-block',
        backgroundColor: '#fff',
        borderRadius: 5,
        width: 600,
        maxWidth: '100%'
      }
    },
    _react2.default.createElement(
      'div',
      { style: {
          display: 'flex',
          backgroundColor: (0, _colorManipulator.fade)(muiTheme.palette.primary1Color, 0.70),
          borderRadius: 4,
          paddingLeft: 10,
          paddingRight: 10
        }
      },
      _react2.default.createElement(
        _FontIcon2.default,
        { style: { marginLeft: 10, marginTop: 12, marginRight: 15 }, className: 'material-icons', color: muiTheme.palette.textColor },
        'search'
      ),
      _react2.default.createElement(_TextField2.default, {
        style: { width: '100%' },
        underlineShow: false,
        onChange: onChange,
        hintText: hintText ? hintText : intl.formatMessage({ id: 'search' })
      })
    )
  );
};

exports.default = (0, _reactIntl.injectIntl)((0, _muiThemeable2.default)()(SearchField));
module.exports = exports['default'];
