// rmw-shell/es/config.js
import { SpotaIcon } from './components/Icons';
import configureStore from './store';
import getMenuItems from './menuItems';
import locales from './locales';
import grants from './grants';
import { themes } from './themes';

var config = {
  firebase_config: {
    apiKey: "AIzaSyBgXyshWBQQjvOCxUHKZnSCuhREqRiU66U",
    authDomain: "spota-2139f.firebaseapp.com",
    databaseURL: "https://spota-2139f.firebaseio.com",
    projectId: "spota-2139f",
    storageBucket: "spota-2139f.appspot.com",
    messagingSenderId: "738533213905"
  },
  firebase_config_dev: {
    apiKey: "AIzaSyBgXyshWBQQjvOCxUHKZnSCuhREqRiU66U",
    authDomain: "spota-2139f.firebaseapp.com",
    databaseURL: "https://spota-2139f.firebaseio.com",
    projectId: "spota-2139f",
    storageBucket: "spota-2139f.appspot.com",
    messagingSenderId: "738533213905"
  },
  firebase_providers: ['google.com', 'facebook.com', 'twitter.com', 'github.com', 'password', 'phone'],
  initial_state: {
    theme: 'dark',
    locale: 'en'
  },
  drawer_width: 256,
  appIcon: SpotaIcon,
  configureStore: configureStore,
  getMenuItems: getMenuItems,
  locales: locales,
  themes: themes,
  grants: grants,
  routes: [],
  onAuthStateChanged: undefined,
  firebaseLoad: function firebaseLoad() {
    return import('./firebase');
  }
};

export default config;
