// rmw-shell/lib/components/Drawer/DrawerHeader.js
'use strict';

exports.__esModule = true;
exports.DrawerHeader = undefined;

var _Avatar = require('material-ui/Avatar');

var _Avatar2 = _interopRequireDefault(_Avatar);

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _IconButton = require('material-ui/IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _List = require('material-ui/List/List');

var _List2 = _interopRequireDefault(_List);

var _ListItem = require('material-ui/List/ListItem');

var _ListItem2 = _interopRequireDefault(_ListItem);

var _Paper = require('material-ui/Paper');

var _Paper2 = _interopRequireDefault(_Paper);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _withAppConfigs = require('../../withAppConfigs');

var _withAppConfigs2 = _interopRequireDefault(_withAppConfigs);

var _reactIntl = require('react-intl');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DrawerHeader = exports.DrawerHeader = function DrawerHeader(_ref) {
  var muiTheme = _ref.muiTheme,
      intl = _ref.intl,
      auth = _ref.auth,
      setAuthMenuOpen = _ref.setAuthMenuOpen,
      fetchUser = _ref.fetchUser,
      dialogs = _ref.dialogs,
      setDialogIsOpen = _ref.setDialogIsOpen,
      appConfig = _ref.appConfig;

  var styles = {
    header: {
      padding: 5
    },
    header_content: {
      padding: 5
    },
    paper: {
      backgroundColor: muiTheme.palette.primary2Color,
      color: muiTheme.palette.alternateTextColor,
      margin: 0,
      padding: 0
    },
    icon: {
      width: 48,
      height: 48,
      top: 4
    }
  };

  var AppIcon = appConfig.appIcon;

  return _react2.default.createElement(
    _Paper2.default,
    { zDepth: 1, style: styles.paper },
    auth.isAuthorised && _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(
        _List2.default,
        null,
        _react2.default.createElement(_ListItem2.default, {
          disabled: true,
          leftAvatar: _react2.default.createElement(_Avatar2.default, { size: 45, src: auth.photoURL, alt: 'person', icon: _react2.default.createElement(
              _FontIcon2.default,
              { className: 'material-icons' },
              'person'
            ) })
        }),
        _react2.default.createElement(_ListItem2.default, {
          primaryText: auth.displayName,
          secondaryText: auth.email,
          rightIconButton: _react2.default.createElement(
            _IconButton2.default,
            { onClick: function onClick() {
                setDialogIsOpen('auth_menu', !dialogs.auth_menu);
              } },
            _react2.default.createElement(
              _FontIcon2.default,
              { className: 'material-icons' },
              dialogs.auth_menu ? 'arrow_drop_up' : 'arrow_drop_down'
            )
          ),
          disableFocusRipple: true,
          style: { backgroundColor: 'transparent' },
          onClick: function onClick() {
            setDialogIsOpen('auth_menu', !dialogs.auth_menu);
          }
        })
      )
    ),
    !auth.isAuthorised && _react2.default.createElement(
      _List2.default,
      null,
      _react2.default.createElement(_ListItem2.default, {
        disabled: true,
        primaryText: intl.formatMessage({ id: 'app_name' }),
        leftAvatar: _react2.default.createElement(AppIcon, { color: muiTheme.palette.accent1Color, style: styles.icon })
      })
    )
  );
};

exports.default = (0, _reactIntl.injectIntl)((0, _muiThemeable2.default)()((0, _withAppConfigs2.default)(DrawerHeader)));
