// rmw-shell/lib/components/Drawer/DrawerContent.js
'use strict';

exports.__esModule = true;
exports.DrawerContent = undefined;

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _withAppConfigs = require('../../withAppConfigs');

var _withAppConfigs2 = _interopRequireDefault(_withAppConfigs);

var _materialUiSelectableMenuList = require('material-ui-selectable-menu-list');

var _reactIntl = require('react-intl');

var _reactRouterDom = require('react-router-dom');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DrawerContent = exports.DrawerContent = function DrawerContent(props, context) {
  var appConfig = props.appConfig,
      dialogs = props.dialogs,
      intl = props.intl,
      match = props.match,
      messaging = props.messaging;


  var handleChange = function handleChange(event, index) {
    var history = props.history,
        responsiveDrawer = props.responsiveDrawer,
        setDrawerOpen = props.setDrawerOpen;


    if (responsiveDrawer.open && index !== undefined) {
      setDrawerOpen(false);
    }

    if (index !== undefined && index !== Object(index)) {
      history.push(index);
    }
  };

  var menuItems = appConfig.getMenuItems(props);

  var handleSignOut = function handleSignOut() {
    var userLogout = props.userLogout,
        setDialogIsOpen = props.setDialogIsOpen,
        appConfig = props.appConfig,
        setDrawerOpen = props.setDrawerOpen;


    appConfig.firebaseLoad().then(function (_ref) {
      var firebaseApp = _ref.firebaseApp;

      firebaseApp.database().ref('users/' + firebaseApp.auth().currentUser.uid + '/connections').remove();
      firebaseApp.database().ref('users/' + firebaseApp.auth().currentUser.uid + '/notificationTokens/' + messaging.token).remove();
      firebaseApp.database().ref('users/' + firebaseApp.auth().currentUser.uid + '/lastOnline').set(new Date());
      firebaseApp.auth().signOut().then(function () {
        userLogout();
        setDrawerOpen(false);
        setDialogIsOpen('auth_menu', false);
      });
    });
  };

  var authItems = [{
    value: '/my_account',
    primaryText: intl.formatMessage({ id: 'my_account' }),
    leftIcon: _react2.default.createElement(
      _FontIcon2.default,
      { className: 'material-icons' },
      'account_box'
    )
  }, {
    value: '/signin',
    onClick: handleSignOut,
    primaryText: intl.formatMessage({ id: 'sign_out' }),
    leftIcon: _react2.default.createElement(
      _FontIcon2.default,
      { className: 'material-icons' },
      'lock'
    )
  }];

  return _react2.default.createElement(
    'div',
    { style: {
        display: 'flex',
        flexDirection: 'column'
      } },
    _react2.default.createElement(_materialUiSelectableMenuList.SelectableMenuList, {
      items: dialogs.auth_menu ? authItems : menuItems,
      onIndexChange: handleChange,
      index: match ? match.path : '/'
    })
  );
};

exports.default = (0, _reactIntl.injectIntl)((0, _muiThemeable2.default)()((0, _reactRouterDom.withRouter)((0, _withAppConfigs2.default)(DrawerContent))));
