// rmw-shell/lib/containers/Users/User.js
'use strict';

exports.__esModule = true;
exports.User = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactIntl = require('react-intl');

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _Activity = require('../../containers/Activity');

var _Activity2 = _interopRequireDefault(_Activity);

var _materialUiResponsiveMenu = require('material-ui-responsive-menu');

var _actions = require('../../store/simpleValues/actions');

var _reactRouterDom = require('react-router-dom');

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _firekitProvider = require('firekit-provider');

var _fireform = require('fireform');

var _fireform2 = _interopRequireDefault(_fireform);

var _reduxForm = require('redux-form');

var _UserForm = require('../../components/Forms/UserForm');

var _UserForm2 = _interopRequireDefault(_UserForm);

var _UserGrants = require('./UserGrants');

var _UserGrants2 = _interopRequireDefault(_UserGrants);

var _UserRoles = require('./UserRoles');

var _UserRoles2 = _interopRequireDefault(_UserRoles);

var _Tabs = require('material-ui/Tabs');

var _Scrollbar = require('../../components/Scrollbar/Scrollbar');

var _Scrollbar2 = _interopRequireDefault(_Scrollbar);

var _materialUiFilter = require('material-ui-filter');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var path = '/users';
var form_name = 'user';

var User = exports.User = function (_Component) {
  _inherits(User, _Component);

  function User() {
    var _temp, _this, _ret;

    _classCallCheck(this, User);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.handleTabActive = function (value) {
      var _this$props = _this.props,
          history = _this$props.history,
          uid = _this$props.uid;


      history.push(path + '/edit/' + uid + '/' + value);
    }, _this.handleAdminChange = function (e, isInputChecked) {
      var _this$props2 = _this.props,
          firebaseApp = _this$props2.firebaseApp,
          match = _this$props2.match;

      var uid = match.params.uid;

      if (isInputChecked) {
        firebaseApp.database().ref('/admins/' + uid).set(true);
      } else {
        firebaseApp.database().ref('/admins/' + uid).remove();
      }
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  User.prototype.componentWillMount = function componentWillMount() {
    this.props.watchList('admins');
  };

  User.prototype.render = function render() {
    var _props = this.props,
        history = _props.history,
        intl = _props.intl,
        muiTheme = _props.muiTheme,
        match = _props.match,
        admins = _props.admins,
        editType = _props.editType,
        setFilterIsOpen = _props.setFilterIsOpen,
        hasFilters = _props.hasFilters,
        firebaseApp = _props.firebaseApp;


    var uid = match.params.uid;
    var isAdmin = false;

    if (admins !== undefined) {
      for (var _iterator = admins, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();;) {
        var _ref;

        if (_isArray) {
          if (_i >= _iterator.length) break;
          _ref = _iterator[_i++];
        } else {
          _i = _iterator.next();
          if (_i.done) break;
          _ref = _i.value;
        }

        var admin = _ref;

        if (admin.key === uid) {
          isAdmin = true;
          break;
        }
      }
    }

    var menuList = [{
      hidden: editType !== 'grants',
      text: intl.formatMessage({ id: 'open_filter' }),
      icon: _react2.default.createElement(
        _FontIcon2.default,
        { className: 'material-icons', color: hasFilters ? muiTheme.palette.accent1Color : muiTheme.palette.canvasColor },
        'filter_list'
      ),
      tooltip: intl.formatMessage({ id: 'open_filter' }),
      onClick: function onClick() {
        return setFilterIsOpen('user_grants', true);
      }
    }];

    return _react2.default.createElement(
      _Activity2.default,
      {
        iconStyleRight: { width: '50%' },
        iconElementRight: _react2.default.createElement(
          'div',
          null,
          _react2.default.createElement(_materialUiResponsiveMenu.ResponsiveMenu, {
            iconMenuColor: muiTheme.palette.canvasColor,
            menuList: menuList
          })
        ),

        onBackClick: function onBackClick() {
          return history.push('/users');
        },
        title: intl.formatMessage({ id: 'edit_user' }) },
      _react2.default.createElement(
        _Scrollbar2.default,
        null,
        _react2.default.createElement(
          _Tabs.Tabs,
          {
            value: editType,
            onChange: this.handleTabActive },
          _react2.default.createElement(
            _Tabs.Tab,
            {
              value: 'profile',
              icon: _react2.default.createElement(
                _FontIcon2.default,
                { className: 'material-icons' },
                'person'
              ) },
            editType === 'profile' && _react2.default.createElement(
              'div',
              { style: { margin: 15, display: 'flex', justifyContent: 'center' } },
              _react2.default.createElement(
                _fireform2.default,
                {
                  firebaseApp: firebaseApp,
                  name: form_name,
                  path: path + '/',
                  onSubmitSuccess: function onSubmitSuccess(values) {
                    history.push('' + path);
                  },
                  onDelete: function onDelete(values) {
                    history.push('' + path);
                  },
                  uid: uid },
                _react2.default.createElement(_UserForm2.default, _extends({
                  handleAdminChange: this.handleAdminChange,
                  isAdmin: isAdmin
                }, this.props))
              )
            )
          ),
          _react2.default.createElement(
            _Tabs.Tab,
            {
              value: 'roles',
              icon: _react2.default.createElement(
                _FontIcon2.default,
                { className: 'material-icons' },
                'account_box'
              ) },
            editType === 'roles' && _react2.default.createElement(_UserRoles2.default, this.props)
          ),
          _react2.default.createElement(
            _Tabs.Tab,
            {
              value: 'grants',
              icon: _react2.default.createElement(
                _FontIcon2.default,
                { className: 'material-icons' },
                'lock'
              ) },
            editType === 'grants' && _react2.default.createElement(_UserGrants2.default, this.props)
          )
        )
      )
    );
  };

  return User;
}(_react.Component);

User.propTypes = process.env.NODE_ENV !== "production" ? {
  history: _propTypes2.default.object,
  intl: _reactIntl.intlShape.isRequired,
  submit: _propTypes2.default.func.isRequired,
  muiTheme: _propTypes2.default.object.isRequired,
  match: _propTypes2.default.object.isRequired,
  admins: _propTypes2.default.array.isRequired
} : {};

var mapStateToProps = function mapStateToProps(state, ownProps) {
  var auth = state.auth,
      intl = state.intl,
      lists = state.lists,
      filters = state.filters;
  var match = ownProps.match;


  var uid = match.params.uid;
  var editType = match.params.editType ? match.params.editType : 'data';

  var _filterSelectors$sele = _materialUiFilter.filterSelectors.selectFilterProps('user_grants', filters),
      hasFilters = _filterSelectors$sele.hasFilters;

  return {
    hasFilters: hasFilters,
    auth: auth,
    uid: uid,
    editType: editType,
    intl: intl,
    admins: lists.admins
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, _extends({ setSimpleValue: _actions.setSimpleValue, change: _reduxForm.change, submit: _reduxForm.submit }, _materialUiFilter.filterActions))((0, _reactIntl.injectIntl)((0, _reactRouterDom.withRouter)((0, _firekitProvider.withFirebase)((0, _muiThemeable2.default)()(User)))));
