// rmw-shell/lib/containers/Users/Users.js
'use strict';

exports.__esModule = true;
exports.Users = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _reactIntl = require('react-intl');

var _List = require('material-ui/List');

var _Divider = require('material-ui/Divider');

var _Divider2 = _interopRequireDefault(_Divider);

var _Avatar = require('material-ui/Avatar');

var _Avatar2 = _interopRequireDefault(_Avatar);

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _reactRouterDom = require('react-router-dom');

var _firekitProvider = require('firekit-provider');

var _reactList = require('react-list');

var _reactList2 = _interopRequireDefault(_reactList);

var _materialUiFilter = require('material-ui-filter');

var _Icons = require('../../components/Icons');

var _Activity = require('../../containers/Activity');

var _Activity2 = _interopRequireDefault(_Activity);

var _Scrollbar = require('../../components/Scrollbar');

var _Scrollbar2 = _interopRequireDefault(_Scrollbar);

var _SearchField = require('../../components/SearchField');

var _SearchField2 = _interopRequireDefault(_SearchField);

var _materialUiResponsiveMenu = require('material-ui-responsive-menu');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var path = 'users';

var Users = exports.Users = function (_Component) {
  _inherits(Users, _Component);

  function Users() {
    var _temp, _this, _ret;

    _classCallCheck(this, Users);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.getProviderIcon = function (provider) {
      var muiTheme = _this.props.muiTheme;

      var color = muiTheme.palette.primary2Color;

      switch (provider.providerId) {
        case 'google.com':
          return _react2.default.createElement(_Icons.GoogleIcon, { color: color });
        case 'facebook.com':
          return _react2.default.createElement(_Icons.FacebookIcon, { color: color });
        case 'twitter.com':
          return _react2.default.createElement(_Icons.TwitterIcon, { color: color });
        case 'github.com':
          return _react2.default.createElement(_Icons.GitHubIcon, { color: color });
        case 'phone':
          return _react2.default.createElement(
            _FontIcon2.default,
            { className: 'material-icons', color: color },
            'phone'
          );
        case 'password':
          return _react2.default.createElement(
            _FontIcon2.default,
            { className: 'material-icons', color: color },
            'email'
          );
        default:
          return undefined;
      }
    }, _this.handleRowClick = function (user) {
      var _this$props = _this.props,
          history = _this$props.history,
          isSelecting = _this$props.isSelecting;

      history.push(isSelecting ? '/' + isSelecting + '/' + user.key : '/' + path + '/edit/' + user.key + '/profile');
    }, _this.renderItem = function (index, key) {
      var _this$props2 = _this.props,
          list = _this$props2.list,
          intl = _this$props2.intl,
          muiTheme = _this$props2.muiTheme;

      var user = list[index].val;

      return _react2.default.createElement(
        'div',
        { key: key },
        _react2.default.createElement(
          _List.ListItem,
          {
            key: key,
            id: key,
            onClick: function onClick() {
              _this.handleRowClick(list[index]);
            },
            leftAvatar: _react2.default.createElement(_Avatar2.default, { style: { marginTop: 10 }, src: user.photoURL, alt: 'person', icon: _react2.default.createElement(
                _FontIcon2.default,
                { className: 'material-icons' },
                'person'
              ) }),
            rightIcon: _react2.default.createElement(
              _FontIcon2.default,
              { style: { marginTop: 22 }, className: 'material-icons', color: user.connections ? muiTheme.palette.primary1Color : muiTheme.palette.disabledColor },
              'offline_pin'
            ) },
          _react2.default.createElement(
            'div',
            { style: { display: 'flex', flexWrap: 'wrap', alignItems: 'stretch' } },
            _react2.default.createElement(
              'div',
              { style: { display: 'flex', flexDirection: 'column', width: 120 } },
              _react2.default.createElement(
                'div',
                null,
                user.displayName
              ),
              _react2.default.createElement(
                'div',
                {
                  style: {
                    fontSize: 14,
                    lineHeight: '16px',
                    height: 16,
                    margin: 0,
                    marginTop: 4,
                    color: muiTheme.listItem.secondaryTextColor
                  } },
                !user.connections && !user.lastOnline ? intl.formatMessage({ id: 'offline' }) : intl.formatMessage({ id: 'online' }),
                ' ',
                !user.connections && user.lastOnline ? intl.formatRelative(new Date(user.lastOnline)) : undefined
              )
            ),
            _react2.default.createElement(
              'div',
              { style: { marginLeft: 20, display: 'flex', flexWrap: 'wrap', alignItems: 'center' } },
              user.providerData && user.providerData.map(function (p, i) {
                return _react2.default.createElement(
                  'div',
                  { key: i, style: { paddingLeft: 10 } },
                  _this.getProviderIcon(p)
                );
              })
            )
          )
        ),
        _react2.default.createElement(_Divider2.default, { inset: true })
      );
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  Users.prototype.componentDidMount = function componentDidMount() {
    var setSearch = this.props.setSearch;


    setSearch('users', '');
    this.props.watchList(path);
  };

  Users.prototype.render = function render() {
    var _this2 = this;

    var _props = this.props,
        list = _props.list,
        muiTheme = _props.muiTheme,
        setSearch = _props.setSearch,
        intl = _props.intl,
        setFilterIsOpen = _props.setFilterIsOpen,
        hasFilters = _props.hasFilters;


    var menuList = [{
      text: intl.formatMessage({ id: 'open_filter' }),
      icon: _react2.default.createElement(
        _FontIcon2.default,
        { className: 'material-icons', color: hasFilters ? muiTheme.palette.accent1Color : muiTheme.palette.canvasColor },
        'filter_list'
      ),
      tooltip: intl.formatMessage({ id: 'open_filter' }),
      onClick: function onClick() {
        setFilterIsOpen('users', true);
      }
    }];

    var filterFields = [{
      name: 'displayName',
      label: intl.formatMessage({ id: 'name' })
    }, {
      name: 'email',
      label: intl.formatMessage({ id: 'email_label' })
    }];

    return _react2.default.createElement(
      _Activity2.default,
      {
        iconStyleLeft: { width: 'auto' },
        iconStyleRight: { width: '100%', textAlign: 'center', marginLeft: 0 },
        iconElementRight: _react2.default.createElement(
          'div',
          { style: { display: 'flex' } },
          _react2.default.createElement(
            'div',
            { style: { width: 'calc(100% - 84px)' } },
            _react2.default.createElement(_SearchField2.default, {
              onChange: function onChange(e, newVal) {
                setSearch('users', newVal);
              },
              hintText: '' + intl.formatMessage({ id: 'search' })
            })
          ),
          _react2.default.createElement(
            'div',
            { style: { position: 'absolute', right: 10, width: 100 } },
            _react2.default.createElement(_materialUiResponsiveMenu.ResponsiveMenu, {
              iconMenuColor: muiTheme.palette.canvasColor,
              menuList: menuList
            })
          )
        ),
        isLoading: list === undefined },
      _react2.default.createElement(
        'div',
        { style: { height: '100%', overflow: 'none', backgroundColor: muiTheme.palette.canvasColor } },
        _react2.default.createElement(
          _Scrollbar2.default,
          null,
          _react2.default.createElement(
            _List.List,
            { id: 'test', ref: function ref(field) {
                return _this2.list = field;
              } },
            _react2.default.createElement(_reactList2.default, {
              itemRenderer: this.renderItem,
              length: list ? list.length : 0,
              type: 'simple'
            })
          )
        )
      ),
      _react2.default.createElement(_materialUiFilter.FilterDrawer, {
        name: 'users',
        fields: filterFields,
        formatMessage: intl.formatMessage
      })
    );
  };

  return Users;
}(_react.Component);

Users.propTypes = process.env.NODE_ENV !== "production" ? {
  users: _propTypes2.default.array,
  intl: _reactIntl.intlShape.isRequired,
  muiTheme: _propTypes2.default.object.isRequired,
  auth: _propTypes2.default.object.isRequired
} : {};

var mapStateToProps = function mapStateToProps(state, ownProps) {
  var lists = state.lists,
      auth = state.auth,
      filters = state.filters;
  var match = ownProps.match;


  var isSelecting = match.params.select ? match.params.select : false;

  var _filterSelectors$sele = _materialUiFilter.filterSelectors.selectFilterProps('companies', filters),
      hasFilters = _filterSelectors$sele.hasFilters;

  var list = _materialUiFilter.filterSelectors.getFilteredList('users', filters, lists[path], function (fieldValue) {
    return fieldValue.val;
  });

  return {
    isSelecting: isSelecting,
    hasFilters: hasFilters,
    list: list,
    auth: auth
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, _extends({}, _materialUiFilter.filterActions))((0, _reactIntl.injectIntl)((0, _muiThemeable2.default)()((0, _firekitProvider.withFirebase)((0, _reactRouterDom.withRouter)(Users)))));
