// rmw-shell/lib/containers/Users/UserRoles.js
'use strict';

exports.__esModule = true;
exports.UserRoles = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactIntl = require('react-intl');

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _actions = require('../../store/simpleValues/actions');

var _reactRouterDom = require('react-router-dom');

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _firekitProvider = require('firekit-provider');

var _List = require('material-ui/List');

var _Divider = require('material-ui/Divider');

var _Divider2 = _interopRequireDefault(_Divider);

var _Avatar = require('material-ui/Avatar');

var _Avatar2 = _interopRequireDefault(_Avatar);

var _Toggle = require('material-ui/Toggle');

var _Toggle2 = _interopRequireDefault(_Toggle);

var _reactList = require('react-list');

var _reactList2 = _interopRequireDefault(_reactList);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var UserRoles = exports.UserRoles = function (_Component) {
  _inherits(UserRoles, _Component);

  function UserRoles() {
    var _temp, _this, _ret;

    _classCallCheck(this, UserRoles);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.handleRoleToggleChange = function (e, isInputChecked, key) {
      var _this$props = _this.props,
          firebaseApp = _this$props.firebaseApp,
          match = _this$props.match;

      var uid = match.params.uid;

      if (isInputChecked) {
        firebaseApp.database().ref('/user_roles/' + uid + '/' + key).set(true);
      } else {
        firebaseApp.database().ref('/user_roles/' + uid + '/' + key).remove();
      }
    }, _this.renderRoleItem = function (i, k) {
      var _this$props2 = _this.props,
          roles = _this$props2.roles,
          user_roles = _this$props2.user_roles,
          match = _this$props2.match;


      var uid = match.params.uid;
      var key = roles[i].key;
      var val = roles[i].val;
      var userRoles = [];

      if (user_roles !== undefined) {
        user_roles.map(function (role) {
          if (role.key === uid) {
            if (role.val !== undefined) {
              userRoles = role.val;
            }
          }
          return role;
        });
      }

      return _react2.default.createElement(
        'div',
        { key: key },
        _react2.default.createElement(_List.ListItem, {
          leftAvatar: _react2.default.createElement(_Avatar2.default, {
            alt: 'person',
            src: val.photoURL,
            icon: _react2.default.createElement(
              _FontIcon2.default,
              { className: 'material-icons' },
              'account_box'
            )
          }),
          rightToggle: _react2.default.createElement(_Toggle2.default, {
            toggled: userRoles[key] === true,
            onToggle: function onToggle(e, isInputChecked) {
              _this.handleRoleToggleChange(e, isInputChecked, key);
            }
          }),
          key: key,
          id: key,
          primaryText: val.name,
          secondaryText: val.description
        }),
        _react2.default.createElement(_Divider2.default, { inset: true })
      );
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  UserRoles.prototype.componentWillMount = function componentWillMount() {
    this.props.watchList('user_roles');
    this.props.watchList('roles');
  };

  UserRoles.prototype.render = function render() {
    var _this2 = this;

    var roles = this.props.roles;


    return _react2.default.createElement(
      'div',
      { style: { height: '100%' } },
      _react2.default.createElement(
        _List.List,
        { style: { height: '100%' } },
        _react2.default.createElement(_reactList2.default, {
          itemRenderer: function itemRenderer(i, k) {
            return _this2.renderRoleItem(i, k);
          },
          length: roles ? roles.length : 0,
          type: 'simple'
        })
      )
    );
  };

  return UserRoles;
}(_react.Component);

UserRoles.propTypes = process.env.NODE_ENV !== "production" ? {
  intl: _reactIntl.intlShape.isRequired,
  muiTheme: _propTypes2.default.object.isRequired,
  match: _propTypes2.default.object.isRequired
} : {};

var mapStateToProps = function mapStateToProps(state, ownProps) {
  var auth = state.auth,
      intl = state.intl,
      lists = state.lists,
      filters = state.filters;
  var match = ownProps.match;


  var uid = match.params.uid;

  return {
    filters: filters,
    auth: auth,
    uid: uid,
    intl: intl,
    user_roles: lists.user_roles,
    roles: lists.roles ? lists.roles : []
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, { setSimpleValue: _actions.setSimpleValue })((0, _reactIntl.injectIntl)((0, _reactRouterDom.withRouter)((0, _firekitProvider.withFirebase)((0, _muiThemeable2.default)()(UserRoles)))));
