// rmw-shell/lib/components/ReduxFormFields/AvatarImageField.js
'use strict';

exports.__esModule = true;
exports.default = undefined;

var _FlatButton = require('material-ui/FlatButton');

var _FlatButton2 = _interopRequireDefault(_FlatButton);

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Avatar = require('../../containers/Avatar');

var _reduxForm = require('redux-form');

var _ImageCropDialog = require('../../containers/ImageCropDialog');

var _reactIntl = require('react-intl');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var AvatarImageField = function (_Component) {
  _inherits(AvatarImageField, _Component);

  function AvatarImageField(props) {
    _classCallCheck(this, AvatarImageField);

    var _this = _possibleConstructorReturn(this, _Component.call(this, props));

    _this.handlePhotoUploadSuccess = function (snapshot) {
      var change = _this.props.change;


      change('photoURL', snapshot.downloadURL);
      _this.setState({ selected_avatar_image: undefined });
    };

    _this.state = {
      selected_avatar_image: undefined
    };
    return _this;
  }

  AvatarImageField.prototype.render = function render() {
    var _this2 = this;

    var _props = this.props,
        altIconName = _props.altIconName,
        disabled = _props.disabled,
        initialized = _props.initialized,
        intl = _props.intl,
        path = _props.path,
        uid = _props.uid;


    return _react2.default.createElement(
      'div',
      { style: { margin: 20 } },
      _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(_reduxForm.Field, {
          name: 'photoURL',
          size: 120,
          component: _Avatar.Avatar,
          icon: _react2.default.createElement(
            _FontIcon2.default,
            {
              className: 'material-icons' },
            altIconName
          ),
          ref: 'photoURL',
          withRef: true
        })
      ),
      _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(_FlatButton2.default, {
          style: { width: '100%' },
          onClick: function onClick() {
            _this2.setState({ selected_avatar_image: 'true' });
          },
          disabled: disabled === true ? true : uid === undefined || !initialized,
          containerElement: 'label',
          primary: true,
          icon: _react2.default.createElement(
            _FontIcon2.default,
            {
              className: 'material-icons' },
            'photo_camera'
          )
        })
      ),
      _react2.default.createElement(_ImageCropDialog.ImageCropDialog, {
        path: path + '/' + uid,
        fileName: 'photoURL',
        onUploadSuccess: function onUploadSuccess(s) {
          _this2.handlePhotoUploadSuccess(s);
        },
        open: this.state.selected_avatar_image !== undefined,
        src: this.state.selected_avatar_image,
        handleClose: function handleClose() {
          _this2.setState({ 'selected_avatar_image': undefined });
        },
        title: intl.formatMessage({ id: 'change_photo' })
      })
    );
  };

  return AvatarImageField;
}(_react.Component);

exports.default = AvatarImageField;


AvatarImageField.propTypes = process.env.NODE_ENV !== "production" ? {
  disabled: _propTypes2.default.bool.isRequired,
  initialized: _propTypes2.default.bool.isRequired,
  uid: _propTypes2.default.string.isRequired,
  intl: _reactIntl.intlShape.isRequired,
  altIconName: _propTypes2.default.string,
  path: _propTypes2.default.string.isRequired
} : {};
module.exports = exports['default'];
