// rmw-shell/lib/containers/Roles/Role.js
'use strict';

exports.__esModule = true;
exports.Role = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _Activity = require('../../containers/Activity');

var _Activity2 = _interopRequireDefault(_Activity);

var _Avatar = require('material-ui/Avatar');

var _Avatar2 = _interopRequireDefault(_Avatar);

var _Dialog = require('material-ui/Dialog');

var _Dialog2 = _interopRequireDefault(_Dialog);

var _Divider = require('material-ui/Divider');

var _Divider2 = _interopRequireDefault(_Divider);

var _fireform = require('fireform');

var _fireform2 = _interopRequireDefault(_fireform);

var _FlatButton = require('material-ui/FlatButton');

var _FlatButton2 = _interopRequireDefault(_FlatButton);

var _FontIcon = require('material-ui/FontIcon');

var _FontIcon2 = _interopRequireDefault(_FontIcon);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _RoleForm = require('../../components/Forms/RoleForm');

var _RoleForm2 = _interopRequireDefault(_RoleForm);

var _Toggle = require('material-ui/Toggle');

var _Toggle2 = _interopRequireDefault(_Toggle);

var _muiThemeable = require('material-ui/styles/muiThemeable');

var _muiThemeable2 = _interopRequireDefault(_muiThemeable);

var _withAppConfigs = require('../../withAppConfigs');

var _withAppConfigs2 = _interopRequireDefault(_withAppConfigs);

var _List = require('material-ui/List');

var _materialUiResponsiveMenu = require('material-ui-responsive-menu');

var _reduxForm = require('redux-form');

var _reactRedux = require('react-redux');

var _reactIntl = require('react-intl');

var _actions = require('../../store/dialogs/actions');

var _firekitProvider = require('firekit-provider');

var _reactRouterDom = require('react-router-dom');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var path = '/roles';
var form_name = 'role';

var Role = exports.Role = function (_Component) {
  _inherits(Role, _Component);

  function Role() {
    var _temp, _this, _ret;

    _classCallCheck(this, Role);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.validate = function (values) {
      var intl = _this.props.intl;

      var errors = {};

      errors.name = !values.name ? intl.formatMessage({ id: 'error_required_field' }) : '';

      return errors;
    }, _this.handleClose = function () {
      var setDialogIsOpen = _this.props.setDialogIsOpen;


      setDialogIsOpen('delete_role', false);
    }, _this.handleDelete = function () {
      var _this$props = _this.props,
          history = _this$props.history,
          match = _this$props.match,
          firebaseApp = _this$props.firebaseApp;

      var uid = match.params.uid;

      if (uid) {
        firebaseApp.database().ref().child(path + '/' + uid).remove().then(function () {
          _this.handleClose();
          history.goBack();
        });
      }
    }, _this.handleGrantToggleChange = function (e, isInputChecked, key) {
      var _this$props2 = _this.props,
          firebaseApp = _this$props2.firebaseApp,
          match = _this$props2.match;

      var uid = match.params.uid;

      if (isInputChecked) {
        firebaseApp.database().ref('/role_grants/' + uid + '/' + key).set(true);
      } else {
        firebaseApp.database().ref('/role_grants/' + uid + '/' + key).remove();
      }
    }, _this.renderGrantItem = function (i, k) {
      var _this$props3 = _this.props,
          role_grants = _this$props3.role_grants,
          match = _this$props3.match,
          intl = _this$props3.intl,
          appConfig = _this$props3.appConfig;


      var uid = match.params.uid;
      var key = i;
      var val = appConfig.grants[i];
      var roleGrants = [];

      if (role_grants !== undefined) {
        role_grants.map(function (role) {
          if (role.key === uid) {
            if (role.val !== undefined) {
              roleGrants = role.val;
            }
          }
          return role;
        });
      }

      return _react2.default.createElement(
        'div',
        { key: key },
        _react2.default.createElement(_List.ListItem, {
          leftAvatar: _react2.default.createElement(_Avatar2.default, {
            alt: 'person',
            src: undefined,
            icon: _react2.default.createElement(
              _FontIcon2.default,
              { className: 'material-icons' },
              'checked'
            )
          }),
          rightToggle: _react2.default.createElement(_Toggle2.default, {
            toggled: roleGrants[val] === true,
            onToggle: function onToggle(e, isInputChecked) {
              _this.handleGrantToggleChange(e, isInputChecked, val);
            }
          }),
          key: key,
          id: key,
          primaryText: intl.formatMessage({ id: 'grant_' + val })
          //secondaryText={val.description}
        }),
        _react2.default.createElement(_Divider2.default, { inset: true })
      );
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  Role.prototype.componentDidMount = function componentDidMount() {
    this.props.watchList('grants');
    this.props.watchList('role_grants');
  };

  Role.prototype.render = function render() {
    var _props = this.props,
        history = _props.history,
        intl = _props.intl,
        dialogs = _props.dialogs,
        setDialogIsOpen = _props.setDialogIsOpen,
        submit = _props.submit,
        muiTheme = _props.muiTheme,
        match = _props.match,
        firebaseApp = _props.firebaseApp,
        appConfig = _props.appConfig;


    var uid = match.params.uid;

    var actions = [_react2.default.createElement(_FlatButton2.default, {
      label: intl.formatMessage({ id: 'cancel' }),
      primary: true,
      onClick: this.handleClose
    }), _react2.default.createElement(_FlatButton2.default, {
      label: intl.formatMessage({ id: 'delete' }),
      secondary: true,
      onClick: this.handleDelete
    })];

    var menuList = [{
      text: intl.formatMessage({ id: 'save' }),
      icon: _react2.default.createElement(
        _FontIcon2.default,
        { className: 'material-icons', color: muiTheme.palette.canvasColor },
        'save'
      ),
      tooltip: intl.formatMessage({ id: 'save' }),
      onClick: function onClick() {
        submit('role');
      }
    }, {
      hidden: uid === undefined,
      text: intl.formatMessage({ id: 'delete' }),
      icon: _react2.default.createElement(
        _FontIcon2.default,
        { className: 'material-icons', color: muiTheme.palette.canvasColor },
        'delete'
      ),
      tooltip: intl.formatMessage({ id: 'delete' }),
      onClick: function onClick() {
        setDialogIsOpen('delete_role', true);
      }
    }];

    return _react2.default.createElement(
      _Activity2.default,
      {
        iconStyleRight: { width: '50%' },
        iconElementRight: _react2.default.createElement(
          'div',
          null,
          _react2.default.createElement(_materialUiResponsiveMenu.ResponsiveMenu, {
            iconMenuColor: muiTheme.palette.canvasColor,
            menuList: menuList
          })
        ),

        onBackClick: function onBackClick() {
          history.goBack();
        },
        title: intl.formatMessage({
          id: this.props.match.params.uid ? 'edit_role' : 'create_role'
        }) },
      _react2.default.createElement(
        'div',
        { style: { margin: 15, display: 'flex' } },
        _react2.default.createElement(
          _fireform2.default,
          {
            firebaseApp: firebaseApp,
            name: form_name,
            path: path + '/',
            validate: this.validate,
            onSubmitSuccess: function onSubmitSuccess(values) {
              history.push('' + path);
            },
            onDelete: function onDelete(values) {
              history.push('' + path);
            },
            uid: this.props.match.params.uid },
          _react2.default.createElement(_RoleForm2.default, _extends({
            grants: appConfig.grants,
            renderGrantItem: this.renderGrantItem
          }, this.props))
        )
      ),
      _react2.default.createElement(
        _Dialog2.default,
        {
          title: intl.formatMessage({ id: 'delete_role_title' }),
          actions: actions,
          modal: false,
          open: dialogs.delete_role === true,
          onRequestClose: this.handleClose },
        intl.formatMessage({ id: 'delete_role_message' })
      )
    );
  };

  return Role;
}(_react.Component);

var mapStateToProps = function mapStateToProps(state) {
  var auth = state.auth,
      intl = state.intl,
      dialogs = state.dialogs,
      lists = state.lists;


  return {
    auth: auth,
    intl: intl,
    dialogs: dialogs,
    role_grants: lists.role_grants
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, { setDialogIsOpen: _actions.setDialogIsOpen, change: _reduxForm.change, submit: _reduxForm.submit })((0, _reactIntl.injectIntl)((0, _reactRouterDom.withRouter)((0, _firekitProvider.withFirebase)((0, _withAppConfigs2.default)((0, _muiThemeable2.default)()(Role))))));
